src/devices/Common/Iot/Device/Common/GeographicPosition.cs

```diff
@@ -0,0 +1,371 @@
+            return strLonRet;
+        }
+
+        private static string GetLatitudeString(double latitude)
+        {
+            object[] args = new object[7];
+
+            GetDegreesMinutesSeconds(latitude, 2, out var normalizedVal, out var deg, out var min, out var sec);
+            string strNorthOrSouth = GetNorthOrSouth(normalizedVal);
+
+            args[0] = deg;
+            args[1] = DegreesSymbol;
+            args[2] = min;
+            args[3] = MinutesSymbol;
+            args[4] = sec.ToString("00.00");
+            args[5] = SecondsSymbol;
+            args[6] = strNorthOrSouth;
+            string strLatRet = string.Format(CultureInfo.InvariantCulture, "{0}{1} {2:00}{3} {4}{5}{6}", args);
+            return strLatRet;
+        }
+
+        object ICloneable.Clone()
+        {
+            return new GeographicPosition(this);
+        }
+
+        /// <summary>
+        /// Creates a copy of this instance
+        /// </summary>
+        /// <returns></returns>
+        public GeographicPosition Clone()
+        {
+            return new GeographicPosition(this);
+        }
+
+        /// <summary>
+        /// Returns true if this instance contains a valid position.
+        /// An invalid position is either when <see cref="Latitude"/> and <see cref="Longitude"/> and <see cref="EllipsoidalHeight"/> are exactly zero,
+        /// when either value is NaN or when the position is out of range.
+        /// </summary>
+        /// <returns>See above</returns>
+        public bool ContainsValidPosition()
+        {
+            if (((Latitude == 0.0) && (Longitude == 0.0)) && (EllipsoidalHeight == 0.0))
+            {
+                return false;
+            }
+
+            if (double.IsNaN(Latitude) || double.IsNaN(Longitude))
+            {
+                return false;
+            }
+
+            if ((Math.Abs(Latitude) > 90.0) || (Math.Abs(Longitude) > 360.0))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns true if the two positions are (almost) equal. This ignores the altitude.
+        /// </summary>
+        /// <param name="position">Position to compare with</param>
+        /// <returns>True if the two positions are closer than about 1cm</returns>
+        public bool EqualPosition(GeographicPosition position)
+        {
+            if (position == null)
+            {
+                return false;
+            }
+
+            bool ret;
+            if ((Math.Abs((position.Longitude - Longitude)) < ComparisonEpsilon) && (Math.Abs(position.Latitude - Latitude) < ComparisonEpsilon))
+            {
+                ret = true;
+            }
+            else
+            {
+                ret = false;
+            }
+
+            return ret;
+        }
+
+        /// <summary>
+        /// Equality comparer. Compares the two positions for equality within about 1cm.
```

Comment:
In this case I think this is fine to pick default value... could this call into the other method though?

Reply:
No, because this one compares all three values (lat, long and height), while `EqualPosition` explicitly only compares lat and long